scripts/dev:update.sh

```diff
@@ -0,0 +1,39 @@
+set -e -x
+
+# Update the SDK in the sample app during SDK development. 
+# Run this script from the root directory of the sample app. 
+
+# check if yq is installed and suggest installing it not. 
+if ! command -v yq &> /dev/null
+then
+    echo "CLI program, yq, not yet installed. This script requires it. Install it: brew install yq"
+    exit
+fi
+
+# change directory to where you are executing the script. 
+RUNNING_DIRECTORY="$(pwd)"
+cd "$RUNNING_DIRECTORY"
+
+# We need to generate a unique filename each time we build to trick yarn to skip cache.
+# https://github.com/yarnpkg/yarn/issues/5357 
+echo "Generating a unique file name for the package to trick yarn to skip cache."
+UNIQUE_PACKAGE_NAME="customerio-reactnative-$(date "+%s").tgz"
+echo "UNIQUE_PACKAGE_NAME: $UNIQUE_PACKAGE_NAME"
+
+echo "Removing existing install of RN SDK"
+rm -rf node_modules/customerio-reactnative
+
+echo "Building RN SDK"
+cd ../../ 
+rm customerio-reactnative*.tgz || true # remove previous builds so that "mv" command renames the file we just created
+
+# We are generating 
+npm pack && mv customerio-reactnative-*.tgz "$UNIQUE_PACKAGE_NAME" 
```

Comment:
Even though I don't think we'll have more `*.tgz` ever in this directory that we don't want to delete, but I would still prefer packing this into a separate folder so it is easier to locate and safer to delete.

Reply:
The `.gitignore` file specifies `.tgz` files in it. Because of this, that makes me think that we consider all `.tgz` files to be temporary and OK to delete any of them. 

That is why this script is destructive to `.tgz` files. Treating them all as temporary. 

I may not longer need to do this after [modifying the script to being a static path instead of dynamic](https://github.com/customerio/customerio-reactnative/pull/177#discussion_r1273490113) so we can come back to this after that's done.